08_cell_line_prediction/nbconverted/download_data.py

```diff
@@ -0,0 +1,353 @@
+    # https://depmap.org/portal/download/
+    ccle_sample_info_df = pd.read_csv(
+        'https://ndownloader.figshare.com/files/35020903',
+        sep=',', index_col=0
+    )
+    os.makedirs(os.path.join(cfg.data_dir, 'ccle'), exist_ok=True)
+    ccle_sample_info_df.to_csv(cfg.ccle_sample_info)
+
+print(ccle_sample_info_df.shape)
+print(ccle_sample_info_df.columns)
+ccle_sample_info_df.iloc[:5, :5]
+
+
+# In[3]:
+
+
+if os.path.isfile(cfg.ccle_expression):
+    ccle_expression_df = pd.read_csv(cfg.ccle_expression, sep=',', index_col=0)
+else:
+    print('Loading expression data from CCLE download page...', file=sys.stderr)
+    # URL from CCLE public download data:
+    # https://depmap.org/portal/download/
+    ccle_expression_df = pd.read_csv(
+        'https://ndownloader.figshare.com/files/34989919',
+        sep=',', index_col=0
+    )
+    os.makedirs(os.path.join(cfg.data_dir, 'ccle'), exist_ok=True)
+    ccle_expression_df.to_csv(cfg.ccle_expression)
+
+ccle_expression_df.index.name = 'DepMap_ID'
+    
+print(ccle_expression_df.shape)
+ccle_expression_df.iloc[:5, :5]
+
+
+# ### Download and process mutation data
+
+# In[4]:
+
+
+if os.path.isfile(cfg.ccle_mutation):
+    ccle_mutation_df = pd.read_csv(cfg.ccle_mutation, sep=',', index_col=0)
+else:
+    print('Loading mutation data from CCLE download page...', file=sys.stderr)
+    # URL from CCLE public download data:
+    # https://depmap.org/portal/download/
+    ccle_mutation_df = pd.read_csv(
+        'https://ndownloader.figshare.com/files/34989940', sep=','
+    )
+    os.makedirs(os.path.join(cfg.data_dir, 'ccle'), exist_ok=True)
+    ccle_mutation_df.to_csv(cfg.ccle_mutation)
+    
+print(ccle_mutation_df.shape)
+print(ccle_mutation_df.columns)
+ccle_mutation_df.iloc[:5, :5]
+
+
+# In[5]:
+
+
+# process mutations into binary matrix
+# https://github.com/greenelab/pancancer/blob/d1b3de7fa387d0a44d0a4468b0ac30918ed66886/scripts/initialize/process_sample_freeze.py#L86
+mutations = [
+    'Frame_Shift_Del',
+    'Frame_Shift_Ins',
+    'In_Frame_Del',
+    'In_Frame_Ins',
+    'Missense_Mutation',
+    'Nonsense_Mutation',
+    'Nonstop_Mutation',
+    'RNA',
+    'Splice_Site',
+    'Translation_Start_Site',
+]
+
+sample_mutations = (ccle_mutation_df
+    .query('Variant_Classification in @mutations')
+    .groupby(['DepMap_ID', 'Chromosome', 'Hugo_Symbol'])
+    .apply(len)
+    .reset_index()
+    .rename(columns={0: 'mutation'})
+)
+
+sample_mutations.iloc[:5, :5]
+
+
+# In[6]:
+
+
+ccle_mutation_binary_df = (sample_mutations
+    .pivot_table(index='DepMap_ID',
+                 columns='Hugo_Symbol',
+                 values='mutation',
+                 fill_value=0)
+    .astype(bool)
+    .astype(int)
+)
+
+print(ccle_mutation_binary_df.shape)
+ccle_mutation_binary_df.iloc[:5, :5]
+
+
+# In[7]:
+
+
+ccle_mutation_binary_df.sum(axis='columns').head()
+
+
+# In[8]:
+
+
+ccle_mutation_binary_df.sum(axis='index').sort_values(ascending=False).head(10)
```

Comment:
Yeah, Greg and I have both looked at TTN a bit in our mutation prediction experiments. It's frequently mutated because it's a huge gene/protein, the longest protein in the genome in terms of amino acids.

It's not really thought to be a strong driver in any cancer types, so Greg was using it as a negative control (and so was I for a while) but there is a bit of predictive signal for TTN mutation and some evidence that it could be weakly involved in some cancers, although it's likely that most of its mutations aren't cancer-functional. A few of the other genes that show up in this list (OBSCN = obscurin, MUC16 = mucin subunit) are also large genes/proteins and probably aren't common cancer drivers either, which is good to keep in mind even though we're mostly focusing on drivers we know we can predict the mutation status of for now.